core/src/main/scala/org/apache/spark/eventhubs/EventHubsConf.scala

```diff
@@ -64,7 +64,7 @@ final class EventHubsConf private (private val connectionStr: String)
 
     val lowerKey = key.toLowerCase
 
-    if (self.get(lowerKey).isDefined) {
+    if (self.get(lowerKey).isDefined && lowerKey != ConnectionStringKey.toLowerCase) {
       logWarning(s"$key has already been set to ${self.get(key).get}. Overwriting with $value")
```

Comment:
don't print `self.get(key).get` - that needs to be removed!

Reply:
Sure. Let me send fix that as well.

Reply:
you can remove `&& lowerKey != ConnectionStringKey.toLowerCase` as well (if you stop logging the value).

Reply:
For a fix, I think we need to log existing and new value for any update as long as config key is not a connection string. For connection string, we will stop logging the connection string. Do you agree?

Reply:
That's fine - if that's what is desired, then you can leave it as is. I thought it'd be better to just never log old/new values and print out a consistent message for all cases.